### PR TITLE
feat(headless,atomic): add support for load-more-products for commerce engine and atomic

### PR DESCRIPTION
**Pull request by olamothe**
* Add support for `fetchMoreProduct` in Headless commerce. New action(s) for search, product listing, recommendations which compute next page based on current number of product displayed and concat the response with existing products.
* Expose those new actions on commerce pagination controller as a `fetchMoreProducts()` function.
* Refactor `atomic-load-more-results` to extract functional components
* Add new `atomic-commerce-load-more-products` component,.
* Mass rename `fetchResults` to `fetchProducts` in commerce sub package, because this was triggering me 😎 

https://coveord.atlassian.net/browse/KIT-3050

### CODE REVIEW

**Review by louis-bompart — APPROVED** [review]

File: packages/headless/src/controllers/commerce/core/pagination/headless-core-commerce-pagination.test.ts [hunk 1]
```diff
@@ -152,8 +154,15 @@ describe('core pagination', () => {
...
 
-    it('dispatches #fetchResultsActionCreator', () => {
-      expect(fetchResultsActionCreator).toHaveBeenCalled();
+    it('dispatches #fetchProductsActionCreator', () => {
+      expect(fetchProductsActionCreator).toHaveBeenCalled();
+    });
+  });
+
+  describe('#fetchMoreProducts', () => {
+    it('dispatches #fetchMoreProducts', () => {
```
Comment (fbeaudoincoveo) [inline packages/headless/src/controllers/commerce/core/pagination/headless-core-commerce-pagination.test.ts:163]:
```suggestion
    it('dispatches #fetchMoreProductsActionCreator', () => {
```

File: packages/headless/src/controllers/commerce/core/pagination/headless-core-commerce-pagination.ts [hunk 2]
```diff
@@ -48,6 +48,11 @@ export interface Pagination extends Controller {
    */
   setPageSize(pageSize: number): void;
 
+  /**
+   * Fetch the next page of products, and append them to the current list of products.
```
Comment (fbeaudoincoveo) [inline packages/headless/src/controllers/commerce/core/pagination/headless-core-commerce-pagination.ts:52]:
```suggestion
   * Fetches the next page of products, and appends them to the current list of products.
```

File: packages/headless/src/features/commerce/recommendations/recommendations-selector.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import {isNullOrUndefined} from '@coveo/bueno';
```
Comment (fbeaudoincoveo) [inline packages/headless/src/features/commerce/recommendations/recommendations-selector.ts:1]: In general, should we test our selectors?

https://redux.js.org/usage/writing-tests#selectors

**Review by fbeaudoincoveo — APPROVED** [review]

File: packages/headless/src/features/commerce/recommendations/recommendations-selector.ts (hunk 3 above)
Comment (olamothe) [reply packages/headless/src/features/commerce/recommendations/recommendations-selector.ts:1]: Yeahhhhhh, I guess we should.

File: packages/atomic/src/components/commerce/atomic-commerce-load-more-products/atomic-commerce-load-more-products.tsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+ * The `atomic-commerce-load-more-products` component allows the user to load additional products if more are available.
+ *
+ * @part container - The container of the component.
+ * @part showing-results - The summary displaying which products are shown and how many are available.
+ * @part highlight - The highlighted number of products displayed and number of products available.
+ * @part progress-bar - The progress bar displaying a percentage of results shown over the total number of products available.
+ * @part load-more-results-button - The "Load more products" button.
+ *
+ * @internal
+ */
```
Comment (fpbrault) [inline packages/atomic/src/components/commerce/atomic-commerce-load-more-products/atomic-commerce-load-more-products.tsx:33]:
```suggestion
/**
 * The `atomic-commerce-load-more-products` component allows the user to load additional products if more are available.
 *
 * @internal
 */
```

**Pull request merged**